### PR TITLE
add subscription as a first screen

### PR DESCRIPTION
**Pull request by madclem**
- in `home/index.js`, I added a condition `showSubsription` to implement the logic
- in `subscription-ad/index.js`, there is this method:
```
onSignIn = () => {
    // Do something here!
  }
```
- let me know if the names make sense too!

### CODE REVIEW

File: src/screens/subscription-ad/index.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import Title from 'app/components/title'
+import Paragraph from 'app/components/paragraph'
+import Carousel, { Dots } from 'app/components/carousel'
+
+class SubscriptionAd extends Component {
+  state = {
+    index: 0
+  }
+
+  // is there a better place for these data?
```
Comment (madclem) [inline src/screens/subscription-ad/index.js]: is there a better practice than adding an array of data like this? @ianmcgregor : Styles are now in the main spritesheets

File: src/screens/subscription-ad/index.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      image: require('./images/family_carousel.png'),
+      style: styles.item__family,
+      title: 'Everyone\'s is invited',
+      text: 'Share the fun and build your Pigzbe network with family and friends *'
+    }
+  ]
+
+  onSignIn = () => {
+    // Do something here!
+  }
```
Comment (madclem) [inline src/screens/subscription-ad/index.js]: do something here

File: src/screens/subscription-ad/styles.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+const paddingSideHeader = isTablet ? 70 : 10
+const paddingTopHeader = isTablet ? 30 : 10
+const normalHeight = 600
+const bigHeight = 750
+const normalPlus = isHeight.biggerThan(normalHeight)
+const bigPlus = isHeight.biggerThan(bigHeight)
+const ratioBig = 1.2
+const ratioSmall = 0.89
+const ratioIpad = 2.159
+
```
Comment (madclem) [inline src/screens/subscription-ad/styles.js]: some crazy ass responsive... let me know if it looks weird on your phone, we can probably tweak the `normalHeight` and `bigHeight`

File: src/utils/is-height/index.js [hunk 4]
```diff
@@ -11,11 +11,11 @@ const smallerOrEqual = (value) => {
...
 
 const biggerThan = (value) => {
-  return height < value
+  return height > value
 }
 
 const biggerOrEqual = (value) => {
-  return height <= value
+  return height >= value
 }
```
Comment (madclem) [inline src/utils/is-height/index.js]: fix stupid mistakes

File: src/components/header/index.js [hunk 5]
```diff
@@ -17,9 +17,9 @@ export default ({
 }) => (
   <View style={[styles.container, scrolled ? styles.scrolled : null]}>
     <View style={[styles.iconContainer, { marginTop: topDistance || 0 }]}>
-      {onBack && (
+      {(onBack || backIcon !== 'back') && ( // sometimes we may want to display an icon here or something, backIcon will be present but not onBack
         <IconButton
-          disabled={loading}
+          disabled={!onBack || loading}
           icon={backIcon}
```
Comment (madclem) [inline src/components/header/index.js]: not used anymore but I left it in case it could be useful

File: src/components/dashboard-clouds/index.js [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
 
-export default () => (
-  <View style={styles.clouds}>
+export default ({ style, transparent }) => (
+  <View style={[styles.clouds, style]}>
     <Image
       style={styles.cloudsImage}
-      source={require('./images/clouds.png')}
+      source={transparent ? images.transparent : images.normal}
       resizeMode='contain'
```
Comment (madclem) [inline src/components/dashboard-clouds/index.js]: clouds now have a transparent equivalent. At term, we could probably remove the other opaque one

File: src/screens/home/index.js [hunk 7]
```diff
@@ -3,17 +3,21 @@ import { connect } from 'react-redux'
 import withLoadExchange from 'app/hocs/with-load-exchange'
 import HomeParent from 'app/screens/home-parent'
 import HomeKids from 'app/screens/home-kids'
+import SubscriptionAd from 'app/screens/subscription-ad'
 import Updates from 'app/screens/updates'
 import { getUserAppVersion } from 'app/selectors'
 
 const Home = props => {
+  const showSubscription = true
   return (
```
Comment (madclem) [inline src/screens/home/index.js]: do something here

File: src/components/carousel/index.js [hunk 8]
```diff
@@ -88,22 +93,24 @@ export default class Carousel extends Component {
...
       dotsStyle,
       prevNextStyle,
       theme = 'blue',
-      numItems
+      numItems,
+      noArrows,
+      noDots
     } = this.props
 
     return (
```
Comment (madclem) [inline src/components/carousel/index.js]: a few new props for the carousel:
- noArrows: to remove arrows
- noDots: to remove the dots
- widthTotal: for a special case, when the width item > window width, the scrollview acts funny

File: src/screens/subscription-ad/index.js (hunk 2 above)
Comment (robotlabs) [reply src/screens/subscription-ad/index.js]: ahahah ok :)

File: src/screens/subscription-ad/index.js (hunk 1 above)
Comment (robotlabs) [reply src/screens/subscription-ad/index.js]: seems ok. what will be cleaner is just to have an external file, and you import it. something like carouselSlides.js. just to reduce the verbosity of the class

File: src/screens/subscription-ad/styles.js [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+  containerItemCarousel: {
+    width: windowW
+  },
+  containerItemCarouselImage: {
+    flexDirection: 'row',
+    justifyContent: 'flex-end'
+  },
+  title: {
+    marginBottom: normalPlus ? (isTablet ? 35 : (bigPlus ? 10 : 18)) : 10,
+    marginTop: normalPlus ? (isTablet ? 25 : (bigPlus ? 28 : 18)) : 5,
```
Comment (robotlabs) [inline src/screens/subscription-ad/styles.js]: what is bigPlus?

Comment (madclem) [reply src/screens/subscription-ad/styles.js]: it's screens that are big+, so we have  small /normal or biggerPlus/bigger or biggerPlus!
`small`, `normal` and `big` are anchored values that we can change if we want to

**Pull request closed without merging**